install_ddev.sh

```diff
@@ -14,26 +12,110 @@ BINOWNER=$(ls -ld /usr/local/bin | awk '{print $3}')
 USER=$(whoami)
 SHACMD=""
 FILEBASE=""
+CURRENT_DIR=$PWD
+
+#Explain what the script does
+printf "
+${GREEN}
+####
+# This script will install everything you need to participate in this sprint.
+#
+# Feel free to first inspect the script before continuing if you like.
+#  -To do this just open it with a text editor
+#
+# It does the following:
+#  -Install Docker for your OS if you don't have it already
+#  -Install ddev by Drud Technology
+#  -Copy required components to ~/Sites/sprint/
+#  -Do a git pull to ensure you have latest commits for core
+#  -Pre-load docker images for the sprint toolkit:
+#    -Drupal 8
+#    -phpmyadmin
+#    -Coud9 IDE
+#    -Thelounge IRC client
+#
+# Press y to continue
+# !!You don't need to hit enter!!.
+####
+${RESET}"
+read -n1 INSTALL
+if [[ ! $INSTALL =~ ^[Yy]$ ]]
+then
+    exit 1
+fi
+
+clear
 
 if [[ "$OS" == "Darwin" ]]; then
     FILEBASE="ddev_macos"
+
+    if ! command -v docker >/dev/null 2>&1; then
+        printf "
+        ${GREEN}
+        ####
+        # Installing docker so that ddev will work.
+        #
+        # Press y to continue
+        # !!You don't need to hit enter!!.
+        #
+        ####
+        ${RESET}"
+        read -n1 SEVEN
+        if [[ ! $SEVEN =~ ^[Yy]$ ]]
+        then
+            exit 1
+        fi
+
+        # Install and open Docker
+        hdiutil attach -nobrowse "${CURRENT_DIR}/docker_installs/Docker.dmg"
+        sleep 10
+        cp -rp /Volumes/Docker/Docker.app /Applications/
+        wait
+        open -a /Applications/Docker.app
+        hdiutil detach /Volumes/Docker
+
+        printf "
+        ${GREEN}
+        ####
+        # Please open Docker preferences and set Memory to 3.0 GiB on the Advanced tab.
```

Comment:
I like this step because it forces the user to remember to start Docker.

Steps necessary:

1. Open Docker.
2. Open Docker preferences 
3. Set memory (if supported by computer hardware)
4. Restart Docker
5. Wait for  Docker to start again
6. Press "y"